affinity/cpp-20/d0796r3.md

```diff
@@ -348,31 +376,32 @@ execution::execution_context execContext(systemResource[0]);
-}
 ```
-*Listing 5: Example of constructing an execution context from an execution resource*
+*Listing 5: Example of constructing an `execution_context` from an `execution_resource`*
 
-When creating an `execution_context` from a given `execution_resource`, the executors and allocators associated with it are bound to that `execution_resource`. For example, when creating an `execution_resource` from a CPU socket resource, all executors associated with the given socket will spawn execution agents with affinity to the socket partition of the system *(Listing 6)*.
+When creating an `execution_context` from a given `execution_resource`, the executors associated with it are bound to that `execution_resource` and any `execution_resource`s encapsulated by it. For example, when creating an `execution_resource` from a CPU socket resource, all executors associated with the given socket will spawn execution agents with affinity to the socket partition of the system *(Listing 6)*.
 
 ```cpp
 auto systemResource = std::this_system::discover_topology();
-// FindASocketResource is a user-defined function that finds a 
-// resource that is a CPU socket in the given resource list
-auto& socket = findASocketResource(systemResource);
-execution_contextC{socket} // Associated with the socket
-auto executor = eC.executor(); // By transitivity, associated with the socket too
-auto socketAllocator = eC.allocator(); // Retrieve an allocator to the closest memory node
-std::vector<int, decltype(socketAllocator)> v1(100, socketAllocator);
-std::generate(par.on(executor), std::begin(v1), std::end(v1), std::rand);
+
+/* find_socket_resource is a user-defined function that finds a resource that is a CPU socket in the given resource list */
+auto socket = find_socket_resource(systemResource);
+
+/* Create an execution_context and executor associated with the CPU socket */
+execution_context context{socket}
+auto executor = context.executor();
+
+/* Create an allocator from the memory resource associate with the GPU socket */
+pmr::polymorphic_allocator<int> alloc{socket.memory_resource()};
+
+pmr::vector<int> vec(100, alloc);
+std::generate(par.on(executor), std::begin(vec), std::end(vec), std::rand);
 ```
-*Listing 6: Example of allocating with affinity to an execution resource*
+*Listing 6: Example of executing and allocating with affinity*
 
-The construction of an `execution_context` on a component implies affinity (where possible) to the given resource. This guarantees that all executors created from that `execution_context` can access the resources and the internal data structures requires to guarantee the placement of the processor.
+The construction of an `execution_context` on an `execution_resource` implies affinity (where possible) to the given resource. This guarantees that all executors created from that `execution_context` can access the resources and the internal data structures requires to guarantee the placement of the processor.
```

Comment:
"data structures requires" -> "data structures required"

Not sure what "guarantee the placement of the processor" means. Has "processor" been defined in this context?

Reply:
"placement of the data" perhaps?

Reply:
Good catch, I believe this wording is describing the guarantee that executors can make about where execution agents run. I've changed this to "binding of execution agents", does this read better?